packages/svelte/src/compiler/phases/3-transform/server/visitors/SnippetBlock.js

```diff
@@ -12,7 +12,8 @@ export function SnippetBlock(node, context) {
 	let fn = b.function_declaration(
 		node.expression,
 		[b.id('$$payload'), ...node.parameters],
-		/** @type {BlockStatement} */ (context.visit(node.body))
+		/** @type {BlockStatement} */ (context.visit(node.body)),
+		node.body.metadata.is_async
```

Comment:
we can revert this, the snippet function never needs to be async

```suggestion
		/** @type {BlockStatement} */ (context.visit(node.body))
```